apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Table/TxnHistoryTable.tsx

```diff
@@ -1,153 +1,199 @@
-import { useEffect, useState } from 'react';
+import { SubGraphResponse } from 'hooks/core/types';
+import { useCallback, useEffect, useState } from 'react';
 import styled from 'styled-components';
+import { fetchGenericSubgraph } from 'utils/subgraph';
+import { TxHistoryFilterType } from '.';
+import { DashboardType } from '../DashboardContent';
+import { format } from 'date-fns';
+import { TableRow, TxnDataTable } from './TxnDataTable';
+import { PaginationControls } from './TxnPaginationControl';
+import env from 'constants/env/local';
 
-type TableRow = {
-  date: string;
-  debit: number;
-  credit: number;
+type Props = {
+  dashboardType: DashboardType;
+  filter: TxHistoryFilterType;
 };
 
-type Props = {
-  filter: string;
+type StrategyType = 'TlcStrategy' | 'DsrBaseStrategy' | 'RamosStrategy' | 'TempleBaseStrategy' | 'All';
+
+type Transactions = {
+  hash: string;
+  amount: string;
+  amountUsd: string;
+  id: string;
+  from: string;
+  to: string;
+  kind: 'Repay' | 'Borrow';
+  timestamp: string;
+}[];
+
+type StrategyTxns = {
+  name: StrategyType;
+  id: string;
+  transactionCount: number;
+  transactions: Transactions;
+}[];
+
+type Meta = {
+  block: {
+    number: number;
+  };
 };
 
-const TxnHistoryTable = ({ filter }: Props) => {
+type FetchTxnsResponse = SubGraphResponse<{ strategies: StrategyTxns; _meta: Meta }>;
+
+const TxnHistoryTable = ({ dashboardType, filter }: Props) => {
   const [currentPage, setCurrentPage] = useState(1);
   const [rowsPerPage, setRowsPerPage] = useState(10);
   const [data, setData] = useState<TableRow[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [blockNumber, setBlockNumber] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
+
+  const restartTable = useCallback(async()=> {
+    setData([]);
+    setCurrentPage(1);
+    setBlockNumber(0);
+    setTotalPages(0);
+  }, []);
+  // Set default values TotalPages & BlockNumber on start
+  useEffect(() => {
+    (async ()=> {
+      await restartTable();
+      const strategyType = dashboardTypeToStrategyType(dashboardType);
+      const strategyQuery = strategyType === 'All' ? `` : `( where: {name: "${strategyType}"} )`;
+      // TODO: workout how to update transactionCount give the filter
+
+      const { data: res } = await fetchGenericSubgraph<FetchTxnsResponse>(
```

Comment:
Added generics to fetch subgraph functions, that way we can have typed responses